internal/pkg/template/workload.go

```diff
@@ -154,6 +160,7 @@ type WorkloadOpts struct {
 	LogConfig   *LogConfigOpts
 	Autoscaling *AutoscalingOpts
 	Storage     *StorageOpts
+	Network     NetworkOpts
```

Comment:
Is the fact that this isn't a pointer going to cause issues with template execution? I'm just calling it out because it clearly bit me, as you fixed some inconsistencies in template functions due to typing. 

I think the main situation this causes problems is if you rely on the existence of the Network member struct to inject the CFN for NetworkOpts.

Reply:
I think it's actually the opposite. The fact that it's not a pointer makes it easy for us to not mess it up, like I can always call `.Network.AssignPublicIP` without having to check if it's `nil` first!

For `Network` i think it makes sense that it's not a pointer because it needs to be always present (the default is public)

Reply:
Updated the `NetworkOpts` struct to take directly CFN fields as input, and changed it to a pointer.

If the input to a `template.Parse*` method is nil, then we call `defaultNetworkOpts` which will set the placement to be public subnets